### PR TITLE
feat(git): add `gupom` and `gupomi` aliases

### PR DESCRIPTION
**Pull request by stevenpitts**
Alias `gupom` to `git pull --rebase origin $(git_main_branch)`.
It is a very common practice for me to rebase off latest remote origin/main once I am finished with a PR, so that the commit history is clean.

## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- Alias `gupom` to `git pull --rebase origin $(git_main_branch)`
- Update README to reflect the new alias

## Other comments:

*Note in this section that I am using `master`/`main`/`$(git_main_branch)` interchangibly in some places, as the PR I am referencing was created before using `git_main_branch` was common practice.*

Also see @hikaru-shindo's PR https://github.com/ohmyzsh/ohmyzsh/pull/8988, which introduces `alias grbom='git rebase origin/master'`.
This is a good suggestion, but it does not cover the full scope of my desires, mainly because it rebases off local origin/master (the version last fetched), not remote origin/master.
I could achieve similar behavior with `gfa && grbom`, but I feel that rebasing off the latest remote master is a common enough practice that a convenient single alias is justified.

CC @hikaru-shindo, @SimenB, @mcornella

### CODE REVIEW

**Comment by makusu2** [comment]
@hikaru-shindo @SimenB @mcornella I see a lot of PRs getting ignored on this repo, is there any guide on how I can increase the changes of my PR getting reviewed?
I can't find anything on the contribution guidelines...

**Comment by SimenB** [comment]
I'm not a maintainer here, so I can't help you

**Comment by makusu2** [comment]
@robbyrussell I see you merged https://github.com/ohmyzsh/ohmyzsh/pull/10445 but not this one, could you please let me know what I can do to get this merged?

**Comment by jmeridth** [comment]
@stevenpitts would you be willing to update your PR to also include `gupomi` that aliases to `git pull --rebase=interactive origin $(git_main_branch)` to allow interactive rebasing?  Would be a great compliment to my open PR #10901.  I don't want to steal your thunder on this solution. If not, I can add to my PR.  Cheers.

**Comment by jmeridth** [comment]
@stevenpitts i feel ya. But the maintainers are active just short handed it seems.

File: plugins/git/git.plugin.zsh
Comment (jmeridth) [inline plugins/git/git.plugin.zsh:303]: Thank you

**Review by jmeridth — APPROVED** [review]

**Review by jmeridth — APPROVED** [review]
@mcornella could you please take a look at this one? Thank you in advance. Cheers.

**Comment by carlosala** [comment]
There are tons of PRs adding aliases for git. That's why we take a long time until git aliases are merged.

**Pull request merged**
